app/controllers/api/grades_controller.rb

```diff
@@ -0,0 +1,49 @@
+module Api
+  class GradesController < ApiController
+    skip_before_action :doorkeeper_authorize!
+
+    def orca_response
+      body = orca_response_params
+      key = body[:key]
+      grade_id = key['grade_id']
+      secret = key['secret']
+      grade = Grade.find_by(id: grade_id)
+      if grade.nil?
+        return render json: { message: "No grade for ID #{grade_id}" },
```

Comment:
In situations like these where you're rejecting responses due to invalid data, and you don't entirely trust the sender not to be spraying your system probing for weaknesses, it's better not to deliver any information here.  Just return status 400 (Error: Bad request), and Audit.log more information instead.